[Cmake] Add autoconf vars to toolchain

fuzzard opened the pull request:
## Description
Adds autoconf build system tools/vars to Toolchain file for cross compile of Apple/Android platforms

## Motivation and context
Enable the ability to cross compile autoconf based internal deps for Apple/Android platforms
Implementation of fstrcmp as an example of the usage.

Currently fstrcmp will build ok with host==target, however if cross compiling, it will fail.
Our cross compile platforms currently are Android/Apple, and they all use a Toolchain.cmake file.
My question is whether adding this into the Toolchain file is the best way to go, or if it should be moved into the core cmake to enable platforms that dont use a toolchain to potentially set the required variables?
Also open to alternative variable naming re the PROJECT_XXXENV variables and the ~~DEPBUILDENV~~ DEP_BUILDENV variable

~~Unfortunately the DEPBUILDENV may need to be added to each step (CONFIGURE, PATCH, BUILD) depending on whats done. I dont know of a way to easily insert automatically, as a single step may have multiple commands as demonstrated by the fstrcmp CONFIGURE_COMMAND. You can chain commands without using the COMMAND to break them, but i feel the separation the COMMAND usage gives makes things clear. In the fstrcmp example, the env vars are only required for the autoreconf and configure steps of the CONFIGURE_COMMAND.
May be able to do some crazy regex matches to insert, but before going down that route, i figure id see if this is acceptable in general first.~~

## How has this been tested?
Build fstrcmp using -DENABLE_INTERNAL_FSTRCMP=ON for ios/osx platforms

## What is the effect on users?
N/A

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
[cmake] Toolchain add autoconf based tools/variables

Adds common autoconf based build system tools/variables to Toolchain for
Android/Apple builds

Commit:
[cmake] FindFstrcmp clean internal BUILD_DEP_TARGET

search for autoconf binary as a requirement
dont use PATCH_COMMAND for autoreconf, just use CONFIGURE_COMMAND

Use vars in toolchain to enable cross compile of autoconf based fstrcmp for Apple/Android
platforms.

wsnipex commented:
maybe BUILD_DEP_TARGET() could inject DEPBUILDENV? And I'd name it DEP_BUILDENV

Commit:
[cmake] BUILD_DEP_TARGET add DEP_BUILDENV to CONFIGURE_COMMAND invocations

Sets environment variables for autoconf based build systems for externalproject_add
build of a dependency

fuzzard commented:
Have renamed to DEP_BUILDENV as suggested.

Have also managed to have it generated inside BUILD_DEP_TARGET, so any dep build that doesnt use cmake_args will have the env setup.

I wont extend it past the CONFIGURE_COMMAND for now, but from memory when i was working something similar up before, pythonmodules needed env vars setup in BUILD_COMMAND. I'll deal with that if/when we need to. Think its best to keep it minimalist for now.

wsnipex reviewed (approved):
looks good!
